custom/lib/User.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Passage\Client;
+
+use OpenAPI\Client\Api\TokensApi;
+use OpenAPI\Client\Configuration;
+use OpenAPI\Client\Api\UsersApi;
+use OpenAPI\Client\Api\UserDevicesApi;
+use OpenAPI\Client\Model\UserInfo as PassageUser;
+use OpenAPI\Client\Model\UpdateUserRequest as UpdateUserArgs;
+use OpenAPI\Client\Model\CreateUserRequest as CreateUserArgs;
+use OpenAPI\Client\Model\WebAuthnDevices;
+use OpenAPI\Client\ApiException;
+
+class User
+{
+    private readonly UsersApi $usersApi;
+    private readonly UserDevicesApi $userDevicesApi;
+    private readonly TokensApi $tokensApi;
+
+    /**
+     * User class for handling operations to get and update user information.
+     */
+    public function __construct(private string $appId, Configuration $config)
+    {
+        $this->appId = $appId;
+        $this->usersApi = new UsersApi(null, $config);
+        $this->userDevicesApi = new UserDevicesApi(null, $config);
+        $this->tokensApi = new TokensApi(null, $config);
+    }
+
+    /**
+     * Get a user's object using their user ID.
+     *
+     * @param string $userId The Passage user ID
+     * @return PassageUser Passage User object
+     * @throws PassageError
+     */
+    public function get(string $userId): PassageUser
+    {
+        try {
+            return $this->usersApi->getUser($this->appId, $userId)->getUser();
+        } catch (ApiException $e) {
+            throw PassageError::fromApiException($e);
+        }
+    }
+
+    /**
+     * Get a user's object using their user identifier.
+     *
+     * @param string $identifier The Passage user email or phone number
+     * @return PassageUser Passage User object
+     * @throws PassageError
+     */
+    public function getByIdentifier(string $identifier): PassageUser
+    {
+        try {
+            $users = $this->usersApi->listPaginatedUsers(
+                $this->appId,
+                limit: 1,
+                identifier: strtolower($identifier)
+            )->getUsers();
+
+            if (count($users) === 0) {
+                throw PassageError::fromApiException(
+                    new ApiException(
+                        null,
+                        404,
+                        null,
+                        "{\"code\":\"user_not_found\",\"error\":\"Could not find user with that identifier.\"}",
+                    ),
```

Comment:
Will this be the expected pattern across SDKs for this case, in major releases?

Just thinking of how we might keep it in sync where needed.

Reply:
The [node](https://github.com/passageidentity/passage-node/blob/1eadf6795756dff135cb8a23126c1458e35a4473/src/classes/PassageError/PassageError.ts#L40) `PassageError` factory is pretty similar with an implementation like [this](https://github.com/passageidentity/passage-node/blob/1eadf6795756dff135cb8a23126c1458e35a4473/src/classes/User/User.ts#L70)

In this SDK the last argument is a bit rough due to the codegen'd signature of `ApiException`.

I'm a bit mixed about the current node implementation because it leaves the extra fields we added empty. I imagine in the major version we'd set the network values in order to throw a fully populated `PassageError` object

Reply:
I see. Part of why I asked is because I think I'd expect this to read as `"Could not find user by identifier."` or `"Could not find user."`

It's just nitpicking copy, but I was curious if this message had been established elsewhere already.

Thanks for the extra context.

Reply:
It might be simpler to standardize on `"User not found."`, aligning with a potential server response. The phrasing is concise enough that further edits are unlikely to be needed.

If that works, passage-node could be updated to match. Since this version of the string is already in the package, updating it would be a nice-to-have for the upcoming passage-node minor release.